lib/WDI5.js

```diff
@@ -134,58 +80,106 @@ module.exports = class WDI5 {
         return this;
     }
 
-    /**
-     * fires a press event on the control
-     * @return {WDI5} this for method chaining
-     */
-    press() {
-        this._fireEvent('press');
-        return this;
-    }
+    // --- private methods ---
 
     /**
-     * sap.m.CheckBox convenience function to check a checkbox
-     * @return {WDI5} this for method chaining
+     * retrieves the webelement id for an array of ui5 control ids
+     * @param {Array} aControls UI5
+     * @param {WebdriverIO.BrowserObject} context
      */
-    check() {
-        this._setProperty('selected', true);
-        this._fireEvent('select', { selected: true });
-        return this;
+    _retrieveElements(aControls, context = this._context) {
+        let aResult = [];
+        // loop through items
+        aControls.forEach((item) => {
+            // item id -> create selector
+            const selector = {
+                wdio_ui5_key: item.id, // plugin-internal, not part of RecordReplay.ControlSelector
+                selector: {
+                    id: item.id
+                }
+            };
+
+            // get WDI5 control
+            aResult.push(context.asControl(selector));
+        });
+
+        return aResult;
     }
 
     /**
-     * sap.m.CheckBox convenience function to uncheck a checkbox
-     * @return {WDI5} this for method chaining
+     * attaches to the instamce of this class the functions givven in the parameter sReplFunctionNames
+     * @param {Array} sReplFunctionNames
      */
-    uncheck() {
-        this._setProperty('selected', false);
-        this._fireEvent('select', { selected: false });
-        return this;
+    _attachControlBridge(sReplFunctionNames) {
+        sReplFunctionNames.forEach(sMethodName => {
+            this[sMethodName] = this._executeControlMethod.bind(this, sMethodName, this._webElement, this._context);
+        });
     }
 
-    /**
-     * sap.m.CheckBox convenience function to toggle a checkboxes' state
-     * @return {WDI5} this for method chaining
-     */
-    toggle() {
-        const bState = this._getProperty('selected');
-        this._setProperty('selected', !bState);
-        this._fireEvent('select', { selected: !bState });
-        return this;
-    }
+    _executeControlMethod(methodName, webElement = this._webElement, context = this._context, ...args) {
```

Comment:
please add some jsdoc - specifically what's happening here.
i'm guessing that theres a diff btw executing an interaction event (press, ...) vs executing a function of a ui5 control...